### PR TITLE
fix marshalling

### PR DESCRIPTION
**Pull request by gpsanant**

### CODE REVIEW

File: merkle_util_test.go [hunk 1]
```diff
@@ -250,6 +251,33 @@ func TestProveWithdrawals(t *testing.T) {
...
+	versionedBeaconState, err := beacon.CreateVersionedState(&oracleState)
+	if err != nil {
+		fmt.Println("error", err)
+	}
+
+	_, err = beacon.MarshalSSZVersionedBeaconState(versionedBeaconState)
+	if err != nil {
+		fmt.Println("error", err)
+	}
+	assert.Nil(t, err, "Error %v failed")
```
Comment (gpsanant) [inline merkle_util_test.go:278]: can you marshal oracleState and make sure the bytes are equal

File: merkle_util_test.go [hunk 2]
```diff
@@ -250,6 +251,33 @@ func TestProveWithdrawals(t *testing.T) {
...
+	if err != nil {
+		fmt.Println("error", err)
+	}
+
+	versionedBeaconState, err := beacon.UnmarshalSSZVersionedBeaconState(oracleStateBytes)
+	if err != nil {
+		fmt.Println("error", err)
+	}
+	assert.Equal(t, versionedBeaconState.Version, spec.DataVersionDeneb, "Version %v failed")
+	assert.Nil(t, err, "Error %v failed")
```
Comment (gpsanant) [inline merkle_util_test.go:265]: can you marshal versionedBeaconState.Deneb and make sure the bytes are equal

File: merkle_util_test.go [hunk 3]
```diff
@@ -250,6 +251,33 @@ func TestProveWithdrawals(t *testing.T) {
 	assert.True(t, flag, "Historical Summary Block Root Proof %v failed")
 }
 
+func TestUnmarshalSSZVersionedBeaconState(t *testing.T) {
+	oracleStateBytes, err := oracleState.MarshalSSZ()
```
Comment (gpsanant) [inline merkle_util_test.go:255]: can we do it for a capella state too

File: merkle_util_test.go (hunk 1 above)
Comment (Sidu28) [reply merkle_util_test.go:278]: done

**Review by Sidu28 — APPROVED** [review]

**Pull request merged**